src/XCCDF/xccdf_session.c

```diff
@@ -486,8 +486,7 @@ int xccdf_session_set_profile_id_by_suffix(struct xccdf_session *session, const
 		return 2;
 	} else {
 		const bool search_result = xccdf_session_set_profile_id(session, full_profile_id);
-		assert(search_result);
-		return 0;
+		return (search_result ? 0 : 1);
```

Comment:
this variable is superfluous, why not just use the result of set_profile_id directly?

Reply:
ok